src/main/java/sim/coffee/Server.java

```diff
@@ -1,14 +1,23 @@
 package sim.coffee;
 
-public class Server implements Runnable {
+public class Server implements Runnable{
+    private SharedQueue queue;
 
-    public Server(SharedQueue orders) {
+    public Server(SharedQueue queue) {
+        this.queue = queue;
     }
 
+    /**
+     * Run only while its not done and queue isn't empty
+     */
     @Override
     public void run() {
-        // TODO Auto-generated method stub
-
+        int queueSize = queue.getCustomerOrder().length;
```

Comment:
Just what Lina said, this should go in the loop and then I think it's all good

Reply:
I think I left it out so that the local var can be used in the while loop's condition. The int is used twice, for the condition + simulation 😅

Reply:
@LinaRietuma mentioning Lina as well

Reply:
Oh wait yeah I overlooked this, the logic is wrong then:
1. `getCustomerOrder` returns an array with the customer order to be served (should influence sleep time)
2. The size of the actual SharedQueue itself is what should be a condition for the loop

Reply:
`getCustomerOrder` inherently has to be in the loop so that it happens repeatedly until the queue is empty and done

Reply:
Could just use isEmpty() in the condition (the way it was before, simplifying isEmpty() to return boolean ‘empty’).

Reply:
> Oh wait yeah I overlooked this, the logic is wrong then:
> 
> 1. `getCustomerOrder` returns an array with the customer order to be served (should influence sleep time)
> 2. The size of the actual SharedQueue itself is what should be a condition for the loop

@kymckay the `getCustomerOrder()` only need to return customer's order size right? It does not need to return the whole order.

Reply:
> Could just use isEmpty() in the condition (the way it was before, simplifying isEmpty() to return boolean ‘empty’).

Yep. Got it. Just previously I did not realise that there was the boolean to start with 😅

Reply:
>@kymckay the getCustomerOrder() only need to return customer's order size right? It does not need to return the whole order.

In future it will need the whole order as we'll need to process them for the output report. For now though I just want to get the producer/consumer model working and merged